test/helpers.go

```diff
@@ -88,19 +88,18 @@ func RandomReader(seed, size int) *bytes.Reader {
 
 // SetupTarTestFixture extracts the tarFile to outputDir.
 func SetupTarTestFixture(t testing.TB, outputDir, tarFile string) {
-	err := System("sh", "-c", `(cd "$1" && tar xzf - ) < "$2"`,
-		"sh", outputDir, tarFile)
+	f, err := os.Open(tarFile)
+	defer f.Close()
 	OK(t, err)
-}
 
-// System runs the command and returns the exit code. Output is passed on to
-// stdout/stderr.
-func System(command string, args ...string) error {
-	cmd := exec.Command(command, args...)
+	cmd := exec.Command("tar", "xzf", "-")
```

Comment:
I'm using tar files for tests in two cases:
* Providing sample repositories e.g. to test the check functions
* Provide test data to backup

The latter case includes symlinks, hard links, device nodes etc., all that must be extracted from the tar and properly set up so that the tests can run. I think it's better to rely on an external `tar` for this, at least for now.